src/khepri_fun.erl

```diff
@@ -337,7 +337,26 @@ to_standalone_fun2(Fun, State) ->
         fun_kept ->
             {Fun, State};
         undefined ->
-            to_standalone_fun3(Fun, State)
+            undefined
+    end.
+
+to_standalone_fun2_with_retry(Fun, State) ->
+    case to_standalone_fun2(Fun, State) of
+        undefined ->
+            Lock = {{khepri_fun, Fun}, self()},
+            try
+                global:set_lock(Lock, [node()]),
```

Comment:
The lock must be acquired before the try/catch because if this call raises an exception, `global:del_lock()` will be called despite no acquired lock.